contracts/token/ERC721/ERC721.sol

```diff
@@ -386,8 +392,12 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
      * Emits an {Approval} event.
      */
     function _approve(address to, uint256 tokenId) internal virtual {
+        address owner = ownerOf(tokenId);
+        if (to == owner) {
+            revert ERC721InvalidOperator(to);
```

Comment:
This check came from https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4405, but I don't know why it was there in the first place? We allow approve to self in both ERC1155 and ERC20, so it doesn't seem like a matter of consistency.

I would just remove it.

Reply:
I thought it was an ERC requirement, if its not we should indeed remove it.

Reply:
I couldn't find it, but please double check.

Reply:
I "thought" (like yesterday). Could not find it either.